test/Events.test.js

```diff
@@ -0,0 +1,326 @@
+const expect = require("chai").expect;
+const EventEmitter = require("events");
+const sinon = require("sinon");
+const { GameState } = require("../entities/GameState");
+const { createPlayer } = require("./testutils");
+const { PropertyService } = require("../entities/PropertyService");
+
+describe("Events", () => {
+  let gameState;
+  let mockUI;
+  let eventBusEmitter;
+  let EVENTS;
+
+  beforeEach(() => {
+    gameState = new GameState();
+    gameState.currentPlayerActions = {
+      END_TURN: {
+        execute: () => true,
+      },
+    };
+    mockUI = {
+      startGame: () => true,
+      startTurn: (player) => false,
+      displayAvailableActions: (actions) => false,
+      prompt: () => true,
+      endTurn: () => true,
+      rollingDice: () => true,
+      rollDiceDisplay: (shouldDisplay) => true,
+      payFineDisplay: (shouldDisplay) => true,
+      endTurnDisplay: (shouldDisplay) => true,
+      diceRollResults: (roll1, roll2) => true,
+      rollNormalDice: () => true,
+      rollJailDice: () => true,
+      caughtSpeeding: () => true,
+      playerMovement: (position) => true,
+      payFine: () => true,
+      passGo: () => true,
+      jail: () => true,
+      unknownAction: () => true
+    };
+    eventBusEmitter = new EventEmitter();
+    EVENTS = require("../entities/Events")(eventBusEmitter, mockUI, gameState);
+    gameState.players = [createPlayer({ name: "player1" })];
+  });
+  afterEach(() => {
+    // Restore the default sandbox here
+    sinon.restore();
+  });
+  it("constructor should return object with associated event listeners", () => {
+    for (property in EVENTS) {
+      expect(typeof EVENTS[property]).to.equal("function");
+    }
+  });
+
+  describe("startGame", () => {
+    it("should emit start turn event", () => {
+      const startTurnSpy = sinon.spy();
+      eventBusEmitter.on("START_TURN", startTurnSpy);
+      EVENTS.START_GAME();
+      expect(startTurnSpy.callCount).to.equal(1);
+    });
+    it("should make ui call", () => {
+      const uiSpy = sinon.spy();
+      mockUI.startGame = uiSpy;
+      EVENTS.START_GAME();
+      expect(uiSpy.callCount).to.equal(1);
+    });
+  });
+  describe("startTurn", () => {
+    it("should reset speeding counter and turn taken flag", () => {
+      gameState.speedingCounter = 1;
+      gameState.turnTaken = true;
+      EVENTS.START_TURN();
+      expect(gameState.speedingCounter).to.equal(0);
+      expect(gameState.turnTaken).to.be.false;
```

Comment:
How d'you feel about switching these to `expect(..).to.equal(false)` ?
I ask because, in this form, the error message is placed inside the `expect`:
```js
expect(gameState.turnTaken, "Turn was taken").to.be.false;
```
Whereas in the suggested format it is at the end:
```js
expect(gameState.turnTaken).to.equal(false, "Turn was taken");
```
I think both desire a bit in terms of readability, but to me the latter is the lesser of two evils

Reply:
Also, iirc, the `.to.be.false` also throws linting errors in some configurations because it is interpreted as a statement rather than an expression

Reply:
will fix that!